tests/test_model.py

```diff
@@ -11,3 +11,44 @@ def test_unknown_enum_values() -> None:
     assert ProgramKey(value) is ProgramKey.UNKNOWN
     assert SettingKey(value) is SettingKey.UNKNOWN
     assert StatusKey(value) is StatusKey.UNKNOWN
+
+
+async def test_options_settings_status_references_at_events() -> None:
+    """Test that options, settings and status enum keys are referenced in events.
+
+    Check that all OptionKey, SettingKey and StatusKey are referenced in EventKey also.
+    """
+    for option_key in OptionKey.__members__.values():
+        assert option_key in EventKey.__members__.values(), (
+            f"OptionKey.{option_key.name} not in EventKey enumeration"
+        )
+    for setting_key in SettingKey.__members__.values():
+        assert setting_key in EventKey.__members__.values(), (
+            f"SettingKey.{setting_key.name} not in EventKey enumeration"
+        )
+    for status_key in StatusKey.__members__.values():
+        assert status_key in EventKey.__members__.values(), (
+            f"StatusKey.{status_key.name} not in EventKey enumeration"
+        )
+    for event_key in EventKey.__members__.values():
+        if ".Option." in event_key.value and event_key not in (
+            # Exceptions
+            EventKey.BSH_COMMON_OPTION_ELAPSED_PROGRAM_TIME,
+            EventKey.BSH_COMMON_OPTION_ESTIMATED_TOTAL_PROGRAM_TIME,
+            EventKey.BSH_COMMON_OPTION_PROGRAM_PROGRESS,
+            EventKey.BSH_COMMON_OPTION_REMAINING_PROGRAM_TIME,
+            EventKey.BSH_COMMON_OPTION_REMAINING_PROGRAM_TIME_IS_ESTIMATED,
+            EventKey.CONSUMER_PRODUCTS_CLEANING_ROBOT_OPTION_PROCESS_PHASE,
```

Comment:
What's the reason for these exceptions? Can we clarify why in the comment?

Reply:
Sure, I can clarify there also.
The reason is because there are no documented statuses for that, I'm not sure why do they use ".Option.", but I think they are only used for events, or at least that's the conclusion I get from API documentation, where these keys are only at Events section and not as an option at Programs and Options section.